src/mds/Server.cc

```diff
@@ -1655,33 +1655,26 @@ void Server::reply_client_request(MDRequestRef& mdr, MClientReply *reply)
 }
 
 
-void Server::encode_empty_dirstat(bufferlist& bl)
+void Server::encode_empty_dirstat(bufferlist& bl, const Session* session)
```

Comment:
It would be better to relocate this to `CDir::encode_empty_dirstat` as a static method. I also think it would be more appropriate to pass `session->info`. So the method should be:

`static void CDir::encode_empty_dirstat(bufferlist& bl, const session_info_t& info)`

Reply:
Also avoid having duplicate encoding code.